de.dlr.sc.virsat.cef.swtbot.test/src/de/dlr/sc/virsat/cef/swtbot/test/CefHierarchyLevelTest.java

```diff
@@ -0,0 +1,340 @@
+		assertThat("There are documents, equipment parametes and child items now", elementConfigurationEquipment.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_EQUIPMENT_PARAMETERS));
+		SwtBotDebugHelper.logCodeLine();
+		
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("Equipment Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		bot.button("Save").click();
+		openEditor(elementConfigurationEquipment);
+		SwtBotDebugHelper.logCodeLine();
+		assertFalse(bot.checkBox("Equipment Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		final int EXPECTED_NUMBER_TREE_CHILDREN_EQUIPMENT = 1; // Documents folder
+		assertThat("Equipment parameters should have been removed", elementConfigurationEquipment.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_EQUIPMENT));
+		
+	}
+	
+	@Test
+	public void testAddRemoveEquipmentParametersMassParametersViaCheckBox() {
+		
+		openEditor(elementConfigurationEquipment);
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("Equipment Mass Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		bot.button("Save").click();
+		openEditor(elementConfigurationEquipment);
+		
+		SwtBotDebugHelper.logCodeLine();
+		assertTrue(bot.checkBox("Equipment Mass Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		final int EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_MATH_PARAMETERS = 3; // Documents folder, equipment parameters, equipment mass parameters
+		elementConfigurationEquipment.expand();
+		assertThat("There are documents, equipment parametes and equipment mass parameters now", elementConfigurationEquipment.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_MATH_PARAMETERS));
+		SwtBotDebugHelper.logCodeLine();
+		
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("Equipment Mass Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		// Does not ask for save because equipment parameters are still there... So no hierarchy change happening
+		openEditor(elementConfigurationEquipment);
+		SwtBotDebugHelper.logCodeLine();
+		assertFalse(bot.checkBox("Equipment Mass Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		final int EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_PARAMETERS = 2; // Documents folder, system parameters
+		assertThat("Equipment parameters should not have been removed, mass paremeters yes", elementConfigurationEquipment.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_PARAMETERS));
+		SwtBotDebugHelper.logCodeLine();
+	}
+	
+	@Test
+	public void testInitialEnabledLevels() {
+		
+		// Tests how CEF levels can be assigned before any level is assigned 
+		
+		// On configuration root level, the only the system CAs should be enabled
+		SwtBotDebugHelper.logCodeLine();
+		assertEnabled(configurationTreeSystem.contextMenu(conceptCefX.getDisplayName()).menu(getAddCommandName(SystemParameters.class)));
```

Comment:
I suggest to refactor it to something like
`assertEnabled(addCommand(configurationTreeSystem, SystemParameters.class)`
to reduce visual noise